src/app-nordigen/services/nordigen-service.js

```diff
@@ -0,0 +1,412 @@
+    ]);
+
+    handleNordigenError(detailedAccount);
+    handleNordigenError(metadataAccount);
+
+    return {
+      ...detailedAccount.account,
+      ...metadataAccount
+    };
+  },
+
+  /**
+   * Retrieve details about a specific Institution
+   * @param institutionId
+   * @throws {InvalidInputDataError}
+   * @throws {InvalidNordigenTokenError}
+   * @throws {AccessDeniedError}
+   * @throws {NotFoundError}
+   * @throws {ResourceSuspended}
+   * @throws {RateLimitError}
+   * @throws {UnknownError}
+   * @throws {ServiceError}
+   * @returns {Promise<import('../nordigen-node.types.js').Institution>}
+   */
+  getInstitution: async (institutionId) => {
+    const response = await client.getInstitutionById(institutionId);
+
+    handleNordigenError(response);
+
+    return response;
+  },
+
+  /**
+   * Extends provided accounts about details of their institution
+   * @param {{accounts: Array<import('../nordigen.types.js').DetailedAccount>, institutions: Array<import('../nordigen-node.types.js').Institution>}} params
+   * @returns {Promise<Array<import('../nordigen.types.js').DetailedAccount&{institution: import('../nordigen-node.types.js').Institution}>>}
+   */
+  extendAccountsAboutInstitutions: async ({ accounts, institutions }) => {
+    const institutionsById = institutions.reduce((acc, institution) => {
+      acc[institution.id] = institution;
+      return acc;
+    }, {});
+
+    return accounts.map((account) => {
+      const institution = institutionsById[account.institution_id] || null;
+      return {
+        ...account,
+        institution
+      };
+    });
+  },
+
+  /**
+   * Returns account transaction in provided dates
+   * @param {import('../nordigen.types.js').GetTransactionsParams} params
+   * @throws {InvalidInputDataError}
+   * @throws {InvalidNordigenTokenError}
+   * @throws {AccessDeniedError}
+   * @throws {NotFoundError}
+   * @throws {ResourceSuspended}
+   * @throws {RateLimitError}
+   * @throws {UnknownError}
+   * @throws {ServiceError}
+   * @returns {Promise<import('../nordigen.types.js').GetTransactionsResponse>}
+   */
+  getTransactions: async ({ accountId, startDate, endDate }) => {
+    const response = await client.getTransactions({
+      accountId,
+      dateFrom: startDate,
+      dateTo: endDate
+    });
+
+    handleNordigenError(response);
+
+    return response;
+  },
+
+  /**
+   * Returns account available balances
+   * @param accountId
+   * @throws {InvalidInputDataError}
+   * @throws {InvalidNordigenTokenError}
+   * @throws {AccessDeniedError}
+   * @throws {NotFoundError}
+   * @throws {ResourceSuspended}
+   * @throws {RateLimitError}
+   * @throws {UnknownError}
+   * @throws {ServiceError}
+   * @returns {Promise<import('../nordigen.types.js').GetBalances>}
+   */
+  getBalances: async (accountId) => {
+    const response = await client.getBalances(accountId);
+
+    handleNordigenError(response);
+
+    return response;
+  }
+};
+
+export const client = {
```

Comment:
It's the only way that works for me to have the possibility to test this class fully. I had a big issue with mocking Nordigen Client lib so after a few hours I decided that I will wrap it in the const which I mocked instead of the lib

Reply:
Could you drop a comment here to indicate that this is mocked in testing and all API access should go through it?

Reply:
added